README.md

```diff
@@ -1,84 +1,143 @@
+| SignatureSchemeAddressProvider | [0xD2b5084E68230D609AEaAe5E4cF7df9ebDd6375A](https://calibration.filfox.info/en/address/0xD2b5084E68230D609AEaAe5E4cF7df9ebDd6375A)   |
+| BlocklockSignatureScheme | [0x62C9CF8Ff30177d8479eDaB017f38017bEbf10C2](https://calibration.filfox.info/en/address/0x62C9CF8Ff30177d8479eDaB017f38017bEbf10C2)   |
+
+> 💡 **Note:** `BlocklockSender.sol` is the only contract developers need to interact with directly when creating timelock requests, as it abstracts the underlying implementation and upgradeability via proxy.
 
+## ⚙️ How It Works
 
-### Smart Contract Addresses
+1. **Data Encryption (off-chain)**
 
-| Contract        | Address | Network          |
-|-----------------|---------|------------------|
-| **✨ BlocklockSender (Proxy)** | [0xfF66908E1d7d23ff62791505b2eC120128918F44](https://calibration.filfox.info/en/address/0xfF66908E1d7d23ff62791505b2eC120128918F44)   | Filecoin Calibration Testnet |
-| BlocklockSender (Impl) | [0x02097463c21f21214499FAa538240029d2e4A220](https://calibration.filfox.info/en/address/0x02097463c21f21214499FAa538240029d2e4A220)   | Filecoin Calibration Testnet |
-| DecryptionSender (Proxy) | [0x9297Bb1d423ef7386C8b2e6B7BdE377977FBedd3](https://calibration.filfox.info/en/address/0x9297Bb1d423ef7386C8b2e6B7BdE377977FBedd3)   | Filecoin Calibration Testnet |
-| DecryptionSender (Impl) | [0xea9111e44D23029945f2E46b2bFf26b04D15bd6F](https://calibration.filfox.info/en/address/0xea9111e44D23029945f2E46b2bFf26b04D15bd6F)   | Filecoin Calibration Testnet |
-| SignatureSchemeAddressProvider | [0xD2b5084E68230D609AEaAe5E4cF7df9ebDd6375A](https://calibration.filfox.info/en/address/0xD2b5084E68230D609AEaAe5E4cF7df9ebDd6375A)   | Filecoin Calibration Testnet |
-| BlocklockSignatureScheme | [0x62C9CF8Ff30177d8479eDaB017f38017bEbf10C2](https://calibration.filfox.info/en/address/0x62C9CF8Ff30177d8479eDaB017f38017bEbf10C2)   | Filecoin Calibration Testnet |
+     Generate the encrypted data (`TypesLib.Ciphertext`) with the dcipher threshold network public key for the decryption at specified block hight, using the [blocklock.js](https://github.com/randa-mu/blocklock-js). 
 
+    The TypeScript library supports the following Solidity types: `uint256`, `int256`, `address`, `string`, `bool`, `bytes32`, `bytes`, `uint256[]`, `address[]`, and `struct`.
+2. **Blocklock Request**
 
-### Using the Solidity Interfaces
+    Interact with the on-chain contract at `blocklock.requestBlocklock()` to create a blocklock request. Submit the encrypted data and specify the chain height for decryption. After request is stored, and a `requestId` is generated for tracking.
+3. **Decryption**
 
-#### Installation
+    Once the specified block number is reached, a callback function (`receiveBlocklock（）`) will be triggered to deliver the decryption key, allowing the unlocking of encrypted data.
```

Comment:
@azixus @najienka @CluEleSsUK an important design decision we need to make clear here is _when_ the decryption key is provided and the callbacks are called. Is it at block N, at N+1 or at N+n for a small n depending on XYZ?

Reply:
Currently it's at `N+1`.